src/pkg/backup/details/testdata/in_deets.go

```diff
@@ -0,0 +1,366 @@
+package testdata
+
+import (
+	"context"
+	"strings"
+	"testing"
+
+	"github.com/alcionai/clues"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/exp/maps"
+
+	"github.com/alcionai/corso/src/internal/kopia"
+	"github.com/alcionai/corso/src/internal/model"
+	"github.com/alcionai/corso/src/internal/streamstore"
+	"github.com/alcionai/corso/src/pkg/backup"
+	"github.com/alcionai/corso/src/pkg/backup/details"
+	"github.com/alcionai/corso/src/pkg/fault"
+	"github.com/alcionai/corso/src/pkg/path"
+)
+
+// ---------------------------------------------------------------------------
+//	location set handling
+// ---------------------------------------------------------------------------
+
+var exists = struct{}{}
+
+type locSet struct {
+	// map [locationRef] map [itemRef] {}
+	// refs may be either the canonical ent refs, or something else,
+	// so long as they are consistent for the test in question
+	Locations map[string]map[string]struct{}
+	Deleted   map[string]map[string]struct{}
+}
+
+func newLocSet() *locSet {
+	return &locSet{
+		Locations: map[string]map[string]struct{}{},
+		Deleted:   map[string]map[string]struct{}{},
+	}
+}
+
+func (ls *locSet) AddItem(locationRef, itemRef string) {
+	if _, ok := ls.Locations[locationRef]; !ok {
+		ls.Locations[locationRef] = map[string]struct{}{}
+	}
```

Comment:
I think this should actually just call `AddLocation` as otherwise there may be an edge case with the following:
1. RemoveLocation(x)
2. AddItem(x, a)
3. // Location x still marked as removed

I'm unsure if that's intentional or not